File: member_service/src/main/java/com/luckytree/member_service/member/adapter/feign/KakaoTokenFeignClient.java
package com.luckytree.member_service.member.adapter.feign;

import com.luckytree.member_service.member.adapter.data.KakaoTokenResponse;
import org.springframework.cloud.openfeign.FeignClient;
import org.springframework.stereotype.Component;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;

@Component
@FeignClient(name = "kakaoFeignClient", url = "${oauth2.kakao.baseUrl}", configuration = KakaoFeignConfiguration.class)
public interface KakaoTokenFeignClient {

    @PostMapping(value = "/oauth/token")
    KakaoTokenResponse getToken(@RequestBody String kakaoTokenRequest);
}

File: auth/src/main/kotlin/com/example/auth/client/feign/kakao/KakaoTokenFeignClient.kt
package com.example.auth.client.feign.kakao

import com.example.auth.config.InternalFeignConfig
import com.example.auth.client.feign.kakao.data.GetKakaoTokenResponse
import org.springframework.cloud.openfeign.FeignClient
import org.springframework.stereotype.Component
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody

@Component
@FeignClient(
    name = "ExternalTokenFeignClient",
    url = "\${oauth2.kakao.baseUrl}",
    configuration = [InternalFeignConfig::class]
)
interface KakaoTokenFeignClient {
    @PostMapping("/oauth/token")
    fun getToken(@RequestBody tokenRequest: String): GetKakaoTokenResponse
}
